src/query/storages/fuse/src/operations/commit.rs

```diff
@@ -452,15 +463,22 @@ impl FuseTable {
 
                     // Check if there is only insertion during the operation.
                     let mut new_segments = latest_snapshot.segments.clone();
-                    let suffix = new_segments
-                        .split_off(latest_snapshot.segments.len() - base_snapshot.segments.len());
-                    if suffix.ne(&base_snapshot.segments) {
-                        abort_operation
-                            .abort(ctx.clone(), self.operator.clone())
-                            .await?;
-                        return Err(ErrorCode::StorageOther(
-                            "mutation conflicts, concurrent mutation detected while committing segment compaction operation",
-                        ));
+                    match MutatorConflictDetector::detect_conflicts(
```

Comment:
https://github.com/datafuselabs/databend/blob/0f4ebb46f151ae627b6b292591027534ad112f96/src/query/storages/fuse/src/operations/commit.rs#L455-L462 can be removed.